### PR TITLE
Support substrate relay

### PR DESCRIPTION
**Pull request by fewensa**
support substrate relay

For comparison and review purpose:

https://github.com/darwinia-network/parity-bridges-common/compare/master..s2s

### CODE REVIEW

File: substrate-relay/docs/Guide.md [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+`chain` 用于在 `substrate-relay` 中添加链, 为了后续搭建链之间的桥做准备
+
+> 因为目前尚未实现链的动态加载, 实际上通过这里配置的链只能是预先设定好的, 自定义链的添加也需要代码的支持.
+
+
+
+添加链
+
+```bash
+$ substrate-relay.exe config config chain add pangolin --host 127.0.0.1 --port 1234 --signer //Alice
```
Comment (HackFisher) [inline substrate-relay/docs/Guide.md:129]:
```suggestion
$ substrate-relay config chain add pangolin --host pangolin-rpc.darwinia.newtork --port 443 --signer //Alice
```

Most user are not using windows, so might be better to remove ".exe" postfix

Comment (HackFisher) [inline substrate-relay/docs/Guide.md:129]:
```suggestion
$ substrate-relay config chain add pangolin --host pangolin-rpc.darwinia.newtork --port 443 --signer //Alice
```

Most user are not using windows, so might be better to remove ".exe" postfix

File: substrate-relay/docs/Guide.md [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+name            host            port            signer
+pangolin        127.0.0.1               1234           //Alice
+```
+
+```bash
+$ substrate-relay.exe config config chain add millau --host 127.0.0.1 --port 5678 --signer //Alice
```
Comment (HackFisher) [inline substrate-relay/docs/Guide.md:135]:
```suggestion
$ substrate-relay config chain add millau --host 127.0.0.1 --port 5678 --signer //Alice
```

File: substrate-relay/docs/Guide.md (hunk 1 above)
Comment (fewensa) [reply substrate-relay/docs/Guide.md]: Done it.

**Comment by HackFisher** [comment]
Please help fix the ci and lint. @fewensa

**Comment by HackFisher** [comment]
@fewensa Exist conflicts.

**Comment by fewensa** [comment]
> 
> 
> @fewensa Exist conflicts.

when [darwinia-network/substrate-subxt#1](https://github.com/darwinia-network/substrate-subxt/pull/1) pass to test, will solve it

**Comment by fewensa** [comment]
don't merge for this time, need merge substrate-relay to bridger workspace

File: provider/darwinia-bridger/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+jsonrpsee-http-client = "0.2.0-alpha.6"
+async-macros = "2.0.0"
+async-trait = "0.1.40"
+dirs = "3.0.1"
+env_logger = "0.7.1"
+etc = "0.1.11"
+futures = "0.3.5"
+log = "0.4.11"
+structopt = "0.3.18"
+substrate-subxt = { version = "0.15.0", git = "https://github.com/darwinia-network/substrate-subxt.git", branch="bump-jsonrpsee-alpha6" }
```
Comment (HackFisher) [inline provider/darwinia-bridger/Cargo.toml:35]: @fewensa Please update this dep.

**Comment by fewensa** [comment]
@HackFisher  ready to merge

**Review by HackFisher — APPROVED** [review]

**Pull request merged**